knowage-vue/src/modules/documentExecution/dashboard/widget/WidgetEditor/WidgetEditorSettingsTab/common/editor/options/WidgetEditorColumnData.vue

```diff
@@ -77,8 +77,12 @@ export default defineComponent({
     created() {},
     methods: {
         onColumnChanged() {
-            // prettier-ignore
-            const forInsert = `[kn-column='${this.selectedColumnName}'${this.row ? ` row='${this.row}'` : ''}${this.aggregation ? ` aggregation='${this.aggregation}'` : ''}${this.precision ? ` precision='${this.precision}'` : ''}${this.format ? ' format' : ''}${this.prefix ? ` prefix='${this.prefix}'` : ''}${this.suffix ? ` suffix='${this.suffix}'` : ''}]`
+            const forInsert =
+                this.widgetModel.type === 'html'
+                    ? `[kn-column='${this.selectedColumnName}'${this.row ? ` row='${this.row}'` : ''}${this.aggregation ? ` aggregation='${this.aggregation}'` : ''}${` precision='${this.precision}'`}${this.format ? ' format' : ''}${this.prefix ? ` prefix='${this.prefix}'` : ''}${
+                          this.suffix ? ` suffix='${this.suffix}'` : ''
+                      }]`
+                    : `${this.prefix ?? ''}[kn-column='${this.selectedColumnName}'${this.row ? ` row='${this.row}'` : ''}${this.aggregation ? ` aggregation='${this.aggregation}'` : ''}${this.precision ? ` precision='${this.precision}'` : ''}${this.format ? ' format' : ''}]${this.suffix ?? ''}`
```

Comment:
I can understand the desired behaviour of this method, but the form is quite difficult to read (ternary in ternary literals).
Is it possible to make it more readable? Even by using rather some if? Thanks

Reply:
Refactored in a more logical way.
Because its a string builder with many checks, it is more complex in nature.

Reply:
Absolutely, but now is more readable. Thank you